pandas/types/cast.py

```diff
@@ -620,6 +623,14 @@ def astype_nansafe(arr, dtype, copy=True):
         # work around NumPy brokenness, #1987
         return lib.astype_intsafe(arr.ravel(), dtype).reshape(arr.shape)
 
+    if dtype.name in ("datetime64", "timedelta64"):
+        msg = ("Passing in '{dtype}' dtype with no frequency is "
+               "deprecated and will raise in a future version. "
+               "Please pass in '{dtype}[ns]' instead.")
+        warnings.warn(msg.format(dtype=dtype.name),
+                      FutureWarning, stacklevel=2)
```

Comment:
I think you meant 5? (same as the one <a href="https://github.com/pandas-dev/pandas/pull/15987#discussion_r111492393">beneath</a>).  Done.